msteams-platform/toolkit/add-resource.md

```diff
@@ -8,81 +8,91 @@ ms.topic: overview
 
-## Add cloud resources using Teams Toolkit
+## Limitation
+
+If you have created SPFx based tab project, you can't add Azure cloud resources.
+
+## Add cloud resources
+
+**You can add cloud resources by the following methods:**
 
-> [!IMPORTANT]
-> You need to provision each environment after you add a resource.
+* To add cloud resources by using Teams Toolkit in Visual Studio Code
+* To add cloud resources by using command palette
 
-1. Open **Microsoft Visual Studio Code**.
-1. Select **Teams Toolkit** from left pane.
-1. In the Teams Toolkit side bar panel, select **Add cloud resources**:
+  > [!NOTE]
+  > You need to provision for each environment, after you have successfully added the resource in your Teams app.
+  
+* **To add cloud resources by using Teams Toolkit in Visual Studio Code:**
 
-    :::image type="content" source="../assets/images/teams-toolkit-v2/manual/add cloudresources.png" alt-text="Add resources":::
+   1. Open **Visual Studio Code**.
+   1. Select **Teams Toolkit** from left panel.
+   1. Select **Add features** under **DEVELOPMENT**.
 
-   You can also open the command palette and enter **Teams: Add cloud resources**:
+        :::image type="content" source="~/assets/images/teams-toolkit-v2/manual/cloud/select-feature-updated.png" alt-text="add feature" border="true":::
 
-    :::image type="content" source="../assets/images/teams-toolkit-v2/manual/addcloud.png" alt-text="add cloud resources":::
+* **To add cloud resources by using command palette:**
 
-1. From the pop-up, select the cloud resources you want to add to your Teams app project:
+   1. Open **command palette**.
+   1. Enter **Teams:Add features**.
+   1. Press **Enter**.
 
-     :::image type="content" source="../assets/images/teams-toolkit-v2/manual/addresources.png" alt-text="add":::
+        :::image type="content" source="~/assets/images/teams-toolkit-v2/manual/cloud/Teams-add-features.png" alt-text="cloud" border="true":::
 
-1. Select **OK**.
+   1. From the pop-up, select the cloud resources to add in your project.
 
-The selected resources are successfully added to your project.
+        :::image type="content" source="~/assets/images/teams-toolkit-v2/manual/cloud/updated-final-cloud.png" alt-text="final" border="true":::
 
-## Add cloud resources using TeamsFx CLI in command window
+## Add cloud resources using TeamsFx CLI
 
-1. Change directory to your **project directory**.
-1. Execute the following command to add different resources in your project:
+* Change directory to your **project directory**.
+* The following table lists the capabilities and required commands:
 
-|Cloud Resource|Command|
-|---------------|----------|
-| Azure function|`teamsfx resource add azure-function --function-name your-func-name`|
-| Azure SQL database|`teamsfx resource add --function-name your-func-name`|
-| Azure API management|`teamsfx resource add azure-apim`|
-| Azure Key Vault|`teamsfx resource add azure-keyvault`|
+  |Cloud Resource|Command|
```

Comment:
The command line interfaces need to be updated:
* Add Azure Function: `teamsfx add azure-function`
* Add SQL database: `teamsfx add azure-sql`

Reply:
Updated as mentioned